test-harness/src/test/groovy/com/netflix/conductor/test/integration/SubWorkflowSpec.groovy

```diff
@@ -474,4 +488,1254 @@ class SubWorkflowSpec extends AbstractSpecification {
         cleanup: "Ensure that changes to the task def are reverted"
         metadataService.updateTaskDef(persistedTask2Definition)
     }
+
+    /**
+     * On a 3-level workflow where all workflows reach FAILED state because of a FAILED task
+     * in the leaf workflow.
+     *
+     * A retry is executed on the root workflow.
+     *
+     * Expectation: The root workflow spawns a NEW mid-level workflow, which in turn spawns a NEW leaf workflow.
+     * When the leaf workflow completes successfully, both the NEW mid-level and root workflows also complete successfully.
+     */
+    def "Test retry on the root in a 3-level subworkflow"() {
+        //region Test setup: 3 workflows reach FAILED state. Task 'integration_task_2' in leaf workflow is FAILED.
+        setup: "Modify task definition to 0 retries"
+        def persistedTask2Definition = workflowTestUtil.getPersistedTaskDefinition('integration_task_2').get()
+        def modifiedTask2Definition = new TaskDef(persistedTask2Definition.name, persistedTask2Definition.description,
+                persistedTask2Definition.ownerEmail, 0, persistedTask2Definition.timeoutSeconds,
+                persistedTask2Definition.responseTimeoutSeconds)
+        metadataService.updateTaskDef(modifiedTask2Definition)
+
+        and: "an existing workflow with subworkflow and registered definitions"
+        metadataService.getWorkflowDef(SIMPLE_WORKFLOW, 1)
+        metadataService.getWorkflowDef(WORKFLOW_WITH_SUBWORKFLOW, 1)
+
+        and: "input required to start the workflow execution"
+        String correlationId = 'retry_on_root_in_3level_wf'
+        def input = [
+                'param1'   : 'p1 value',
+                'param2'   : 'p2 value',
+                'subwf'    : WORKFLOW_WITH_SUBWORKFLOW,
+                'nextSubwf': SIMPLE_WORKFLOW]
+
+        when: "the workflow is started"
+        def rootWorkflowId = workflowExecutor.startWorkflow(WORKFLOW_WITH_SUBWORKFLOW, 1,
+                correlationId, input, null, null, null)
+
+        then: "verify that the workflow is in a RUNNING state"
+        with(workflowExecutionService.getExecutionStatus(rootWorkflowId, true)) {
+            status == Workflow.WorkflowStatus.RUNNING
+            tasks.size() == 1
+            tasks[0].taskType == 'integration_task_1'
+            tasks[0].status == Task.Status.SCHEDULED
+        }
+
+        when: "poll and complete the integration_task_1 task"
+        def pollAndCompleteTask = workflowTestUtil.pollAndCompleteTask('integration_task_1', 'task1.integration.worker', ['op': 'task1.done'])
+
+        then: "verify that the 'integration_task_1' was polled and acknowledged"
+        verifyPolledAndAcknowledgedTask(pollAndCompleteTask)
+
```

Comment:
Please check workflow state here and that the subworkflow task has been SCHEDULED.